File: app/src/main/java/network/oversee/buttin/SplashActivity.java
package network.oversee.buttin;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.WindowManager;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import network.oversee.buttin.R;

import network.oversee.buttin.utils.SharedObjects;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.github.inflationx.viewpump.ViewPumpContextWrapper;


public class SplashActivity extends AppCompatActivity {

    private static final int SPLASH_TIME_OUT = 3000;
    @BindView(R.id.txtVersionName) TextView txtVersionName;
    SharedObjects sharedObjects ;

    private FirebaseAuth firebaseAuth;

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(ViewPumpContextWrapper.wrap(newBase));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);

        //Get Firebase auth instance
        firebaseAuth = FirebaseAuth.getInstance();

        sharedObjects = new SharedObjects(SplashActivity.this);

        ButterKnife.bind(this);

        txtVersionName.setText(getString(R.string.version,SharedObjects.getVersion(SplashActivity.this)));

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {

                Intent intentLogin;

                if (firebaseAuth.getCurrentUser() != null) {
                    if (checkIfEmailVerified()){
                        intentLogin = new Intent(SplashActivity.this, MainActivity.class);
                        intentLogin.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
                        startActivity(intentLogin);
                        finish();
                    }else{
                        intentLogin = new Intent(SplashActivity.this, IntroActivity.class);
                        intentLogin.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
                        startActivity(intentLogin);
                        finish();
                    }
                } else {
                    intentLogin = new Intent(SplashActivity.this, IntroActivity.class);
                    intentLogin.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
                    startActivity(intentLogin);
                    finish();
                }
            }
        }, SPLASH_TIME_OUT);
    }

    private boolean checkIfEmailVerified() {
        FirebaseUser user = firebaseAuth.getCurrentUser();
        if (user.isEmailVerified()) {
            // user is verified
            return true;
        } else {
            // email is not verified
            // NOTE: don't forget to log out the user.
            firebaseAuth.signOut();
            return false;
        }
    }
}

File: app/src/main/java/download/crossally/apps/hizkiyyah/SplashActivity.kt
package download.crossally.apps.hizkiyyah

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.WindowManager
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.coordinatorlayout.widget.CoordinatorLayout
import butterknife.BindView
import butterknife.ButterKnife
import com.google.firebase.auth.FirebaseAuth

import download.crossally.apps.hizkiyyah.utils.SharedObjects
import io.github.inflationx.viewpump.ViewPumpContextWrapper

class SplashActivity : AppCompatActivity() {
    @JvmField
    @BindView(R.id.txtVersionName)
    var txtVersionName: TextView? = null
    var sharedObjects: SharedObjects? = null
    private var firebaseAuth: FirebaseAuth? = null
    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(ViewPumpContextWrapper.wrap(newBase))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        //window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)

        //Get Firebase auth instance
        firebaseAuth = FirebaseAuth.getInstance()
        sharedObjects = SharedObjects(this@SplashActivity)
        ButterKnife.bind(this)
        //setResult(Activity.RESULT_OK)
        txtVersionName!!.text = getString(R.string.version, SharedObjects.getVersion(this@SplashActivity))
        val content: RelativeLayout = findViewById(R.id.contSplash)
        content.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_STABLE or
                View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or
                View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
        Handler().postDelayed({
            val intentLogin: Intent
            if (firebaseAuth!!.currentUser != null) {
                if (checkIfEmailVerified()) {
                    intentLogin = Intent(this@SplashActivity, MainActivity::class.java)
                    intentLogin.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
                    startActivity(intentLogin)
                    finish()
                } else {
                    intentLogin = Intent(this@SplashActivity, IntroActivity::class.java)
                    intentLogin.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
                    startActivity(intentLogin)
                    finish()
                }
            } else {
                intentLogin = Intent(this@SplashActivity, IntroActivity::class.java)
                intentLogin.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
                startActivity(intentLogin)
                finish()
            }
        }, SPLASH_TIME_OUT.toLong())
    }

    private fun checkIfEmailVerified(): Boolean {
        val user = firebaseAuth!!.currentUser
        return if (user!!.isEmailVerified) {
            // user is verified
            true
        } else {
            // email is not verified
            // NOTE: don't forget to log out the user.
            firebaseAuth!!.signOut()
            false
        }
    }

    companion object {
        private const val SPLASH_TIME_OUT = 3000
    }
}